### ISSUE TITLE
Requests are not getting deallocated

### ISSUE THREAD

**Issue by Mattijah** [dropbox/SwiftyDropbox#295]
I am not 100% sure if I'm getting this right, but it looks like requests are not being properly deallocated. Example:

```swift
let client = DropboxClientsManager.authorizedClient

client?.files
   .download(path: "/xyz)")
   .response { response, error in
      callAnotherCompletionHandler(response, error) // not required
   }
```

The snippet above creates `DownloadRequestMemory`, which references `ApiRequest` which in this case also keeps a reference to my `callAnotherCompletionHandler`. None of these seem to be released when the request is finished.

The problem seem to lie in `DownloadRequestMemory:response(queue, completionHandler)` > `request.setCompletionHandler`, which strongly references the `self` (`DownloadRequestMemory`) - *I have not investigated this any further. It's just my first thought.

**Comment by greg-db** [comment]
Thanks for writing this up. I'll ask the team to look into this.

**Comment by greg-db** [comment]
This should be fixed in the latest version, v7.0.1.
